test.js

```diff
@@ -224,7 +224,7 @@ describe('when connection is established', () => {
 
 		request.on('socket', socket_ => {
 			socket_.once('connect', () => {
-				assert.equal(socket_.timeout, 100);
+				assert.equal(isNaN(socket.timeout) ? socket._idleTimeout : socket.timeout, 100);
```

Comment:
Use a helper function named `socketTimeout` so you don't have to repeat the logic all over and use `Number.isNaN`.

Reply:
`Number.isNaN()` won't work if the passed value is not of type `number`. I have used `isNaN` to check for `undefined`, so replacing with `in` operator which makes more sense in this case.

Reply:
That's a better check, indeed.